src/events/EventChain.ts

```diff
@@ -31,18 +33,47 @@ export default class EventChain {
 		return crypto.verifyEventChainId(DERIVED_ID_VERSION, id, Binary.fromBase58(this.id));
 	}
 
-	public add(event: Event): EventChain {
-		if (!event.previous) event.previous = this.latestHash;
+	public add(event: Event): EventChain
+	public add(partialChain: EventChain): EventChain
+	public add(input: Event|EventChain): EventChain {
+		if (input instanceof EventChain)
+			this._addChain(input);
+		else
+			this._addEvent(input);
+
+		return this;
+	}
+
+	public has(hash: IBinary): boolean {
+		return !!this.events.find(event => event.hash.hex === hash.hex);
+	}
+
+	private _addEvent(event: Event): void {
+		if (!event.previous.hex) event.previous = this.latestHash;
```

Comment:
This should be `if (!event.previous)`. If previous is not set, you can't get the hex.